src/Application/Messenger/Handler/GiftAidResultHandler.php

```diff
@@ -20,14 +20,28 @@ public function __construct(
     public function __invoke(Messages\Donation $donationMessage): void
     {
         $this->logger->info(sprintf(
-            'Donation ID %s Gift Aid claim failure reported by ClaimBot',
+            'Donation ID %s Gift Aid claim result returned by ClaimBot',
             $donationMessage->id,
         ));
 
         /** @var Donation $donation */
         $donation = $this->donationRepository->findOneBy(['uuid' => $donationMessage->id]);
 
-        $donation->setTbgGiftAidRequestFailedAt(new \DateTime());
+        if ($donationMessage->responseSuccess === false) {
```

Comment:
`if (!$donationMessage->responseSuccess)`

Reply:
@alihejazi this is not the same. We need to explicitly ensure that it is `false`, not `null`.